### ISSUE TITLE
An aggregated parameter that compares the flow at a node and another parameter

### ISSUE THREAD

**Issue by BaptisteFrancois** [pywr/pywr#716]
Hi,
I want to create a parameter that would return the minimum value between a constant (i.e., the capacity of a penstock) and the river flow flowing by the entrance of the penstock.

The way I tried to implement this is the following (parameters' section): 
 ```json
"flow_by_the_pit_entrance":{
    "type": "aggregated",
    "agg_func": "sum",
    "parameters": [
        "river_1",
        "river_2"
	]
},
"infiltration_flow_to_pit": {
    "comment": "minimum between 17 cfs (33.72 AFday) and the flow by the pit entrance",
    "type": "aggregated",
    "agg_func": "min",
    "parameters": [
            "flow_by_the_pit_entrance",
            33.72
	]
},

 ```

The obvious issue here is that 'river_1' and 'river_2' are not parameters but nodes. I actually cannot figure out how to create a parameters, which could be used in the 'min' function of the 'infiltration_flow_to_pit' parameter, that would take values equal to the sum of the flows passing through several nodes (e.g., river_1 and river_2). 

Am I missing something here?

**Comment by jetuk** [comment]
The reason you can't do this is, as you point out, that `river_1` and `river_2` are nodes not parameters. There are two approaches depending on whether you need this "infiltration_flow_to_pit" to be calculated using (1) the current or (2) previous time-steps's flows on the nodes.

Option (1) is more difficult because it will require embedding the min function in to the linear programme, which I am unsure if you can do without a MILP - see our previous discussion on this in #702.

Option (2) is easier, but basically means the parameter's calculation today uses yesterday's flows. However, this means we need a new parameter that passes through the flow from a node. We've done something similar with the `DeficitParameter` - in fact this would new parameter would be a simpler version of that. 

In summary (1) is pretty difficult I think, and (2) I can write a new parameter for you to try but it will be using the previous flows in the calculation.

Does that help?

**Comment by BaptisteFrancois** [comment]
Thanks @jetuk 
I think the option 2 is totally fine for now. 
Let me know how I can help (if needed) to write this new parameter.

**Comment by jetuk** [comment]
The other option, I guess, is to extend the functionality of the `AggregatedParameter` to directly support adding nodes. However, I think the advantage of a "shim" parameter that exposes the flow (or volume) of a node is that it can then be used anywhere in the existing code that a parameter can be used.

**Comment by jetuk** [comment]
@BaptisteFrancois see #724 for the new parameter. I think it does what we discussed here.

**Comment by BaptisteFrancois** [comment]
@jetuk this is great. 
I guess I have to update my pywr package to make use of it new parameter, right? 
I am currently using 1.0.0 version installed with conda.
